File: client/common/finalize_bets.go
package common

type finalizeBets struct {
	agencyId uint16
}

func NewFinalizeBets(agencyId uint16) Message {
	return &finalizeBets{agencyId: agencyId}
}

func (m *finalizeBets) GetMessageCode() MessageCode {
	return FINALIZE_BETS
}

func (m *finalizeBets) SendToLotteryCentral(protocol LotteryAgencyProtocol) error {
	return protocol.SendFinalizeBetsMessage(m.agencyId)
}

func (m *finalizeBets) SendToAgency(agency LotteryAgency) error {
	return nil
}

File: server/common/messages/finalize_bets.py
from ..interfaces.message import Message
from ..constants.message_codes import MessageCode
from ..interfaces.protocol import Protocol


class FinalizeBets(Message):
    def __init__(self, agency_id: int):
        self.agency_id = agency_id

    def get_message_code(self) -> MessageCode:
        return MessageCode.FINALIZE_BETS

    def send_to_lottery_central(self, lottery_central):
        lottery_central.finalize_bets(self.agency_id)

    def send_to_agency(self, protocol: Protocol):
        raise NotImplementedError("FinalizeBets is not sent to agencies")
